[receiver/awss3receiver]: Add ingest progress notifications via OpAMP (#33047)
open-telemetry/opentelemetry-collector-contrib: receiver/awss3receiver/README.md

        s3_bucket: "mybucket"
        s3_prefix: "trace"
        s3_partition: "minute"
```

## Notifications
The receiver can send notifications of ingest progress to an OpAmp server using the custom message capability of 
"org.opentelemetry.collector.receiver.awss3" and message type "TimeBasedIngestStatus".
The format of the notifications is a JSON object with the following fields:

| Field             | Description                                                                     |
|:------------------|:--------------------------------------------------------------------------------|
| `telemetry_type`  | The type of telemetry being ingested. One of "traces", "metrics", or "logs".    |
| `ingest_status`   | The status of the data ingestion. One of "ingesting", "failed", or "completed". |
| `start_time`      | The time to start retrieving data in RFC3339 format.                            |
| `stop_time`       | The time to stop retrieving data in RFC3339 format.                             |
| `ingest_time`     | The time of the data currently being ingested in RFC3339 format.                |

> | `failure_message` | Error message if `ingest_status` is "failed".                                   |

Reviewer: Is there a spec for this as part of opAmp or maybe entities?

Author: There is no spec for this and that is the reason why I'm using a Custom Capability.

The "ingesting" status is sent at the beginning of the ingest process before data has been retrieved for the specified time.

> If during the processing of the data an error occurs a status message with `ingest_status` set to "failed" status with 
> the time of the data being ingested when the failure occurred.
> If the ingest process completes successfully a status message with `ingest_status` set to "completed" is sent.

Reviewer 2: Docs writer here. Just a quick edit for clarity. Thanks!

```suggestion
If an error occurs during the processing of the data, a status message with `ingest_status` set to "failed" is sent, along with the time the ingestion failure occurred.
If the ingest process completes successfully, a status message with `ingest_status` set to "completed" is sent.
```
